packages/flutter/test/material/carousel_theme_test.dart

```diff
@@ -0,0 +1,245 @@
+              padding: padding,
+              shape: shape,
+              overlayColor: overlayColor,
+              itemExtent: 100,
+              children: <Widget>[SizedBox(width: 100, height: 100)],
+            ),
+          ),
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+    expect(find.byType(CarouselView), findsOneWidget);
+
+    final Finder paddingFinder = find.descendant(
+      of: find.byType(CarouselView),
+      matching: find.byWidgetPredicate(
+        (Widget widget) => widget is Padding && widget.child is Material,
+      ),
+    );
+
+    expect(paddingFinder, findsOneWidget);
+    final Padding paddingWidget = tester.widget<Padding>(paddingFinder);
+    final Material material = paddingWidget.child! as Material;
+
+    final InkWell inkWell = tester.widget<InkWell>(
+      find.descendant(of: find.byType(CarouselView), matching: find.byType(InkWell)),
+    );
+
+    expect(paddingWidget.padding, padding);
+    expect(material.color, backgroundColor);
+    expect(material.elevation, elevation);
+    expect(material.shape, shape);
+    expect(inkWell.overlayColor, overlayColor);
+  });
+
+  testWidgets('CarouselViewTheme can override Theme.carouselViewTheme', (
+    WidgetTester tester,
+  ) async {
+    const Color globalBackgroundColor = Color(0xfffffff1);
+    const Color globalOverlayColor = Color(0xff000000);
+    const double globalElevation = 5.0;
+    const EdgeInsets globalPadding = EdgeInsets.all(10.0);
+    const OutlinedBorder globalShape = RoundedRectangleBorder();
+
+    const Color localBackgroundColor = Color(0xffff0000);
+    const Color localOverlayColor = Color(0xffffffff);
+    const double localElevation = 10.0;
+    const EdgeInsets localPadding = EdgeInsets.all(15.0);
+    const OutlinedBorder localShape = RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(10)),
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+          carouselViewTheme: const CarouselViewThemeData(
+            backgroundColor: globalBackgroundColor,
+            overlayColor: MaterialStatePropertyAll<Color>(globalOverlayColor),
+            elevation: globalElevation,
+            padding: globalPadding,
+            shape: globalShape,
+          ),
+        ),
+        home: const Scaffold(
+          body: Center(
+            child: CarouselViewTheme(
+              data: CarouselViewThemeData(
+                backgroundColor: localBackgroundColor,
+                overlayColor: MaterialStatePropertyAll<Color>(localOverlayColor),
+                elevation: localElevation,
+                padding: localPadding,
+                shape: localShape,
+              ),
+              child: CarouselView(
+                itemExtent: 100,
+                children: <Widget>[SizedBox(width: 100, height: 100)],
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
```

Comment:
Is this pumpAndSettle necessary? If so, would just a `pump` work? Same question for the above usages of pumpAndSettle.

Not a huge deal but I try to be strict about frame pumping where possible.

Reply:
No, It's not required. I have removed and pushed the change.